File: hotel-service/hotel-service.API/basemodel/response/builder/BaseResponseBuilder.cs
namespace hotel_service.API.basemodel.response.builder;

public class BaseResponseBuilder
{
    public static BaseResponse<T> Build<T>(bool isSuccess, T data)
    {
        return new BaseResponse<T>(isSuccess, data);
    }

    public static BaseResponse<T> Build<T>(bool isSuccess, string message)
    {
        return new BaseResponse<T>(isSuccess, message);
    }

    public static BaseResponse<T> Build<T>(bool isSuccess, ErrorResponse error)
    {
        return new BaseResponse<T>(isSuccess, error);
    }

}

File: booking-service/src/main/java/com/example/bookingservice/common/base/response/builer/BaseResponseBuilder.java
package com.example.bookingservice.common.base.response.builer;


import com.example.bookingservice.common.base.response.BaseResponse;
import com.example.bookingservice.common.base.response.Error;

public class BaseResponseBuilder {

    public static <T>BaseResponse<T> build(boolean isSuccess, T data) {
        return new BaseResponse<T>(isSuccess,data);
    }

    public static <T>BaseResponse<T> build(boolean isSuccess, String message) {
        return new BaseResponse<>(isSuccess,message);
    }

    public static <T>BaseResponse<T> build(boolean isSuccess, Error error) {
        return new BaseResponse<>(isSuccess,error);
    }

}
